### DISCUSSION TITLE
Please provide a package for Raspbian Bullseye 64bit?

### DISCUSSION THREAD

**Discussion by cragginstylie** [PiSupply/PiJuice#805] [Q&A] [answered]
I have purchased numerous PiJuice units, thus far only connected to RPi 3B+ boards. All working great BTW! But, I have a new project which requires more CPU/Mem than an RPi 3 has to offer, so I've migrated over to RPi 4 and I'm using the latest official Raspbian Bullseye arm64 release. Initially dismayed that there wasn't a pre-built package for this platform yet, especially considering how long RPi 4 has been out. But, I was relieved to see your maintainers provided a Debian package builder script. However, while attempting to acquire the prerequisite tools (stdeb, dh-systemd, & debhelper), I discovered the dh-systemd package is not available for _any_ arm systems, 32-bit or 64-bit. I would prefer to stick with your product to avoid re-engineering my power supply portion of my project. Can you help? Any advice? TIA

**Comment by tvoverbeek** [comment]
In the control files for pijuice-base and pijuice-gui remove the dh-systemd dependency.
See this commit: https://github.com/PiSupply/PiJuice/commit/ce0e97d3e9e97b0fee56ce5c368df817eb5b1343
Then it should build on Bullseye

**Reply by cragginstylie** [reply]
Hi, and thanks for the quick response! Unfortunately, it doesn't build:

`pi@tgrctl2:~/Downloads/PiJuice/Software/Source $ ./pckg-pijuice.sh
usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
   or: setup.py --help [cmd1 cmd2 ...]
   or: setup.py --help-commands
   or: setup.py cmd --help

error: invalid command 'bdist_deb'
dpkg-buildpackage: info: source package pijuice-base
dpkg-buildpackage: info: source version 1.7
dpkg-buildpackage: info: source distribution unstable
dpkg-buildpackage: info: source changed by Milan Neskovic <milan@pi-supply.com>
dpkg-buildpackage: info: host architecture arm64
 dpkg-source --before-build .
 fakeroot debian/rules clean
dh clean --with python3 --with systemd --buildsystem=pybuild
dh: error: Compatibility levels before 5 are no longer supported (level 1 requested)
make: *** [debian/rules:5: clean] Error 255
dpkg-buildpackage: error: fakeroot debian/rules clean subprocess returned exit status 2
mv: cannot move 'deb_dist' to 'deb_dist_base/deb_dist': Directory not empty
usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
   or: setup.py --help [cmd1 cmd2 ...]
   or: setup.py --help-commands
   or: setup.py cmd --help

error: invalid command 'bdist_deb'
cp: cannot create directory './deb_dist/pijuice-gui-1.7/debian/': No such file or directory
./pckg-pijuice.sh: line 23: cd: ./deb_dist/pijuice-gui-1.7: No such file or directory
`
I haven't got time to dig into it further atm, so hoping you are easily able to shed some light. If you need me to run some tests or try some things later when I get back home, please let me know. Happy to assist in any way.

**Comment by tvoverbeek** [comment, marked as answer]
Here  a way you get it  working on Raspberry Pi OS bullseye 64-bit.
I downloaded and installed the 64-bit OS and applied all updates.

- Get the 1.7 packages (pijuice-base_1.7_all.deb and pijuice-gui_1.7_all.deb) from github: https://github.com/PiSupply/PiJuice/tree/master/Software/Install
- Before installing pijuice-base manually install the required python3-urwid package: `sudo apt install python3-urwid`
- Install pijuice-base with dpkg from the directory where you downloaded it: `sudo dpkg -i ./pijuice-base_1.7_all.deb`.

After this the pijuice service (a pure python3 implementation) should already be running.
Check with `systemctl status pijuice`
The only problem is that the 32-bit programs (/usr/bin/pijuice_cli and ./usr/bin/pijucieboot) do not work
Use multiarch to solve this:
- `sudo dpkg --add-architecture armhf`
- Now you can install libc6 for armhf: `sudo apt install libc6:armhf`

Both pijuice_cli and pijuiceboot should work now.
If the gui is needed install pijuice-gui_1.7_all.deb.

**Reply by cragginstylie** [reply]
That worked! Thanks a bunch!

**Comment by tvoverbeek** [comment]
SInce today (Feb 11) the 1.8 packages are available in the raspberry pi repositories. They install on both 32-bit and 64-bit and on both buster and bullseye.
For 64-bit no multi-arch is needed.
